### PR TITLE
add deprecation for watchOS

### LINKED ISSUE #10195: [Firebase Database] Database calls do not work on watchOS 9
### Description

The FirebaseDatabase SDK does not work on watchOS 9. I have been having issues with websockets in my own products, and I believe this issue might be related to that. I am using FirebaseAuth just fine, and can make rest calls to the database with Alamofire. 

### Reproducing the issue

Try to read/write from the database on watchOS 9

### Firebase SDK Version

9.5.0

### Xcode Version

13.4.1

### Installation Method

CocoaPods

### Firebase Product(s)

Database

### Targeted Platforms

watchOS

### Relevant Log Output

```shell
Don't have physical access to a watchOS 9 device at the moment, and the issue doesn't show up on emulators.
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

PODS:
  - Alamofire (5.6.2)
  - FirebaseAppCheck (9.5.0):
    - FirebaseCore (~> 9.0)
    - GoogleUtilities/Environment (~> 7.7)
    - PromisesObjC (~> 2.1)
  - FirebaseAuth (9.5.0):
    - FirebaseCore (~> 9.0)
    - GoogleUtilities/AppDelegateSwizzler (~> 7.7)
    - GoogleUtilities/Environment (~> 7.7)
    - GTMSessionFetcher/Core (< 3.0, >= 1.7)
  - FirebaseCore (9.5.0):
    - FirebaseCoreDiagnostics (~> 9.0)
    - FirebaseCoreInternal (~> 9.0)
    - GoogleUtilities/Environment (~> 7.7)
    - GoogleUtilities/Logger (~> 7.7)
  - FirebaseCoreDiagnostics (9.5.0):
    - GoogleDataTransport (< 10.0.0, >= 9.1.4)
    - GoogleUtilities/Environment (~> 7.7)
    - GoogleUtilities/Logger (~> 7.7)
    - nanopb (< 2.30910.0, >= 2.30908.0)
  - FirebaseCoreInternal (9.5.0):
    - "GoogleUtilities/NSData+zlib (~> 7.7)"
  - FirebaseCrashlytics (9.5.0):
    - FirebaseCore (~> 9.0)
    - FirebaseInstallations (~> 9.0)
    - GoogleDataTransport (< 10.0.0, >= 9.1.4)
    - GoogleUtilities/Environment (~> 7.7)
    - nanopb (< 2.30910.0, >= 2.30908.0)
    - PromisesObjC (~> 2.1)
  - FirebaseInstallations (9.5.0):
    - FirebaseCore (~> 9.0)
    - GoogleUtilities/Environment (~> 7.7)
    - GoogleUtilities/UserDefaults (~> 7.7)
    - PromisesObjC (~> 2.1)
  - GoogleDataTransport (9.2.0):
    - GoogleUtilities/Environment (~> 7.7)
    - nanopb (< 2.30910.0, >= 2.30908.0)
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleUtilities/AppDelegateSwizzler (7.7.0):
    - GoogleUtilities/Environment
    - GoogleUtilities/Logger
    - GoogleUtilities/Network
  - GoogleUtilities/Environment (7.7.0):
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleUtilities/Logger (7.7.0):
    - GoogleUtilities/Environment
  - GoogleUtilities/Network (7.7.0):
    - GoogleUtilities/Logger
    - "GoogleUtilities/NSData+zlib"
    - GoogleUtilities/Reachability
  - "GoogleUtilities/NSData+zlib (7.7.0)"
  - GoogleUtilities/Reachability (7.7.0):
    - GoogleUtilities/Logger
  - GoogleUtilities/UserDefaults (7.7.0):
    - GoogleUtilities/Logger
  - GTMSessionFetcher/Core (2.1.0)
  - nanopb (2.30909.0):
    - nanopb/decode (= 2.30909.0)
    - nanopb/encode (= 2.30909.0)
  - nanopb/decode (2.30909.0)
  - nanopb/encode (2.30909.0)
  - NWWebSocket (0.5.2)
  - PromisesObjC (2.1.1)
  - SwiftyStoreKit (0.16.1)

DEPENDENCIES:
  - Alamofire
  - FirebaseAppCheck
  - FirebaseAuth
  - FirebaseCore
  - FirebaseCrashlytics
  - NWWebSocket
  - SwiftyStoreKit

SPEC REPOS:
  trunk:
    - Alamofire
    - FirebaseAppCheck
    - FirebaseAuth
    - FirebaseCore
    - FirebaseCoreDiagnostics
    - FirebaseCoreInternal
    - FirebaseCrashlytics
    - FirebaseInstallations
    - GoogleDataTransport
    - GoogleUtilities
    - GTMSessionFetcher
    - nanopb
    - NWWebSocket
    - PromisesObjC
    - SwiftyStoreKit

SPEC CHECKSUMS:
  Alamofire: d368e1ff8a298e6dde360e35a3e68e6c610e7204
  FirebaseAppCheck: 84e58ced2b43249bb75529da1c33193b71e9c3c6
  FirebaseAuth: 741918f1fa62ed23246a509279f91d42d79a713f
  FirebaseCore: 25c0400b670fd1e2f2104349cd3b5dcce8d9418f
  FirebaseCoreDiagnostics: 17cbf4e72b1dbd64bfdc33d4b1f07bce4f16f1d8
  FirebaseCoreInternal: 50a8e39cae8abf72d5145d07ea34c3244f70862b
  FirebaseCrashlytics: d20fa38bb8c88cb0f1c9211286bc23ab58c1b464
  FirebaseInstallations: 41f811b530c41dd90973d0174381cdb3fcb5e839
  GoogleDataTransport: 1c8145da7117bd68bbbed00cf304edb6a24de00f
  GoogleUtilities: e0913149f6b0625b553d70dae12b49fc62914fd1
  GTMSessionFetcher: ffbb25ec00ebcb5201adab0a56d808f6f1902d9f
  nanopb: b552cce312b6c8484180ef47159bc0f65a1f0431
  NWWebSocket: 21f0c73639815da3272862c912275b26102aa80c
  PromisesObjC: ab77feca74fa2823e7af4249b8326368e61014cb
  SwiftyStoreKit: 6b9c08810269f030586dac1fae8e75871a82e84a

PODFILE CHECKSUM: bc259dea40d518c905f5b56c82dbfb8316383e71

COCOAPODS: 1.11.3


```

</details>

### LINKED ISSUE #10272: Deprecate RTDB APIs for watchOS 9
### Description

RTDB is not currently functional with watchOS 9. See #10195

### Reproducing the issue

_No response_

### Firebase SDK Version

10.0.0

### Xcode Version

14.0.0

### Installation Method

Swift Package Manager

### Firebase Product(s)

Database

### Targeted Platforms

watchOS

### Relevant Log Output

_No response_

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

### PR DESCRIPTION
**Pull request by morganchen12**
Addresses, but #10195 is not resolved.

Fixes #10272

### CODE REVIEW

File: FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h [hunk 1]
```diff
@@ -26,7 +26,13 @@ NS_ASSUME_NONNULL_BEGIN
  * by calling `Database.database()`. To access a location in the database and
  * read or write data, use `FIRDatabase.reference()`.
  */
-NS_SWIFT_NAME(Database)
+NS_SWIFT_NAME(Database) __attribute__((
+    availability(
+        watchos, introduced = 6.0, deprecated = 9.0,
+        message = "Socket connections are not supported on watchOS. "
```
Comment (paulb777) [inline FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h]: "Socket connections are not supported on watchOS 9.0. "

File: FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h [hunk 2]
```diff
@@ -26,7 +26,13 @@ NS_ASSUME_NONNULL_BEGIN
...
+                  "Use the REST API instead. See "
```
Comment (paulb777) [inline FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h]: "Use the Firebase Database REST API instead. See "

**Review by paulb777 — APPROVED** [review]
A bit more clarity, perhaps?

File: FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h (hunk 1 above)
Comment (morganchen12) [reply FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h]: Done.

File: FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h (hunk 2 above)
Comment (morganchen12) [reply FirebaseDatabase/Sources/Public/FirebaseDatabase/FIRDatabase.h]: Done.

**Pull request merged**